sdk/identity/identity/src/credentials/interactiveBrowserCredential.browser.ts

```diff
@@ -147,14 +153,35 @@ export class InteractiveBrowserCredential implements TokenCredential {
   ): Promise<AccessToken | null> {
     const { span } = createSpan("InteractiveBrowserCredential-getToken", options);
     try {
-      if (!this.msalObject.getAccount()) {
-        await this.login();
+      const currentAccounts = this.msalObject.getAllAccounts();
```

Comment:
If you do not have a multiple accounts use case, you can use `getAccount` too! FYI

Reply:
Thanks! I'll take a look.

Reply:
Seems like MSAL Browser's PublicClientApplication doesn't have `getAccount`, but:

```ts
    getAccountByHomeId(homeAccountId: string): AccountInfo | null;
    getAccountByLocalId(localId: string): AccountInfo | null;
    getAccountByUsername(userName: string): AccountInfo | null;
```

Specially on reload, I'm currently (in this PR) trying to match them by the client ID and the tenant ID, this way:

```ts
    const accounts = this.msalObject.getAllAccounts();

    const account = accounts.find((account) => {
      const claims: any = account.idTokenClaims;
      return (
        claims &&
        claims.aud === this.clientId && // Same target client ID
        account.tenantId === this.tenantId // Same target tenant ID
      );
    });
```

What do you think of this approach?

Reply:
Can you not use `homeAccountId` which you can save in your original response? That would be the better cleaner approach.

Reply:
I just pushed an update without this code around getAllAccounts. I realized there wasn't a real scenario in which I wouldn't have this.account after I initially authenticated.